File: api/src/main/java/api/api/Order/Order.java
package api.api.Order;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "orders")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Order {
    @Id
    private String id;
    private String dateFrom;
    private String dateTo;
    private Number totalPrice;
    private String user;
    private String item;
    private String orderType;
    @CreatedDate
    private LocalDateTime createdAt;
    @LastModifiedDate
    private LocalDateTime modifiedAt;
}

File: client/src/types/order.ts
export enum ORDER_TYPES {
    RENT = 'RENT',
    BUY = 'BUY',
}

export enum ORDER_TYPES_LABEL {
    RENT = 'Аренда',
    BUY = 'Покупка',
}

export interface Order {
    id: string;
    orderType: string;
    user: string;
    dateFrom?: string;
    dateTo?: string;
    item: string;
    phone: string;
    totalPrice: number;
    createdAt?: string;
    updatedAt?: string;
}

export interface OrderBody extends Omit<Order, 'id'> {}
